### ISSUE TITLE
mod_http_upload has started returning error 400 bad-request

### ISSUE THREAD

**Issue by rspq** [esl/MongooseIM#1400]
MongooseIM version: mongooseim:2.1.0beta2
Installed from: Package (docker)
Erlang/OTP version: 18.3.4

We had to update our server to to get mod_push and since doing so our S3 uploads have broken. Clients now get error 400: Bad Request when they request an upload slot. We've got the log level as high as possible (5) but nothing informative appears there.

SENT:
`<iq to='upload.HOST' from='rs@HOST/BLAH' id='BLAH' type='get'><request xmlns='urn:xmpp:http:upload:0'><filename>32defdfb-4e43-4395-ace0-293a0dfc31a9</filename><size>3651449</size><content-type>application/octet-stream</content-type></request></iq>`

RECV'D: 
`<iq from='upload.HOST' to='rs@HOST/BLAH' id='BLAH' type='error'><request xmlns='urn:xmpp:http:upload:0'><filename>32defdfb-4e43-4395-ace0-293a0dfc31a9</filename><size>3651449</size><content-type>application/octet-stream</content-type></request><error code='400' type='modify'><bad-request xmlns='urn:ietf:params:xml:ns:xmpp-stanzas'/></error></iq>`

What could cause this to happen? Is there any logging we can enable anywhere to get greater insight into what's going on?

**Comment by rspq** [comment]
I resolved this to the use of the namespace "urn:xmpp:http:upload:0" instead of "urn:xmpp:http:upload". Whether Smack or MongooseIM is at fault I cannot say, but it certainly doesn't work.

**Comment by kzemek** [comment]
Hi @rspq 

This looks like Smack's fault. `urn:xmpp:http:upload:0` refers to [XEP-0363: HTTP File Upload v0.3.x](https://xmpp.org/extensions/xep-0363.html) which specifies that slot request should be in the form:

```xml
<iq to='upload.HOST' from='rs@HOST/BLAH' id='BLAH' type='get'>
    <request xmlns='urn:xmpp:http:upload:0'
             filename='32defdfb-4e43-4395-ace0-293a0dfc31a9'
             size='3651449'
             content-type='application/octet-stream' />
</iq>
```

The module also supports [the v0.2.5 version](https://xmpp.org/extensions/attic/xep-0363-0.2.5.html) which is identified by namespace `urn:xmpp:http:upload` and which uses subelements of `<request/>` rather than attributes in slot request.

**Comment by kzemek** [comment]
Looks like it was fixed in Smack 4.2: https://github.com/igniterealtime/Smack/commit/858ba3f82a2068bc7c1dcfeceaba1838ed315d82
